src/core/linter-rules/no-unused-vars.js

```diff
@@ -0,0 +1,85 @@
+// @ts-check
+/**
+ * Linter rule "no-unused-vars".
+ *
+ * Checks that an variable is used if declared (the first use is treated as
+ * declaration).
+ */
+import LinterRule from "../LinterRule.js";
+import { lang as defaultLang } from "../l10n.js";
+import { norm } from "../utils.js";
+
+const name = "no-unused-vars";
+
+const meta = {
+  en: {
+    description: "Variable was defined, but never used.",
+    howToFix:
+      "Add a `data-ignore-unused` attribute to the `<var>`, or consider using `<em>`/ `<code>`.",
+    help: "See developer console.",
+  },
+};
+// Fall back to english, if language is missing
+const lang = defaultLang in meta ? defaultLang : "en";
+
+/**
+ * @param {*} _
+ * @param {Document} doc
+ * @return {import("../LinterRule").LinterResult}
+ */
+function linterFunction(_, doc) {
+  const offendingElements = [];
+
+  /**
+   * Check if a <section> contains a `".algorithm"`
+   *
+   * The selector matches:
+   * ``` html
+   * <section><ul class="algorithm"></ul></section>
+   * ```
+   * The selector does not match:
+   * ``` html
+   * <section><div><ul class="algorithm"></ul></div></section>
```

Comment:
What’s the rationale for this one?

Reply:
The selector gets complex (match section only if section contains algorithm, but not any subsection). Looking at specs, most `.algorithm` were in directly section, not wrapped in other elements. Might go the imperative way if selector gets too complex and we want to support it.